API 3 of 3: Refactor, Simplify endpoints

R2ZER0 opened the pull request:
This PR was originally part 3/3 of #175 

This PR:
* refactors the Org API out of the "experimental" API module into it's own "org" module (suggestions for a better name welcome)
*   Adds the `djangorestframework-dataclasses` DRF helper, to make Serializers out of dataclasses as well as Django models
* Simplify by removing the individual grant endpoint (and so removing `self` ref links from grants)
* "Simplifies" the Org endpoint fields a bit, with the philosophy of only including things we have a schema/serialiser for, and not just throwing out all the data we have w/o consideration:
  - add `name` to top-level Org object
  - add a serialiser for Funder/Recipient grant aggregate stats
  - remove additional_data that's not planned to be output
  - don't duplicate the `id` in Org sub-objects
  
Tests are in separate PR https://github.com/ThreeSixtyGiving/datastore/pull/183

R2ZER0 committed:
api: Remove GrantDetailView

To simplify the API, remove redundant single grant detail view.
The other views display all the same data.

R2ZER0 committed:
api: Add name to top-level Org model

R2ZER0 committed:
api: Serializer for Funder/Recipient aggregate data

R2ZER0 committed:
api: Remove unused import

R2ZER0 committed:
api: Refactor, into own module

* Move the new Org API into it's own API submodule
* Make use of dataclass serializer library

R2ZER0 committed:
tests: Add tests for Org API

michaelwood commented on datastore/api/experimental/api.py:
```diff
@@ -1,19 +1,17 @@
 import db.models as db
-import api.experimental.models as api_experimental
-from api.experimental import serializers
+from . import serializers
```
Prefer explicit/absolute imports for readability/stability of any refactoring

R2ZER0 committed:
api: Raise 404 on GrantsView if Org doesn't exist

R2ZER0 committed:
tests: Add test for Organisation.exists

R2ZER0 committed:
tests: Fix org_api recipient grants test

Fix a bug in this test missed due toprevious GrantsMadeView behaviour of returning empty list instead of 404.

R2ZER0 committed:
tests: Add Org API tests for non-existent Org behaviour

R2ZER0 committed:
api: Fix bug in OrgDetail for non-existent Org

The OrgDetail view wasn't properly handling object-not-found errors.

R2ZER0 committed:
tests: Fix example OrgIDs to be valid OrgIDs

Example OrgIDs should start with prefix XE-EXAMPLE

R2ZER0 committed:
api: Use absolute imports

Replace relative imports with absolute imports

R2ZER0 committed:
tests: Add tests for Organisation.get()
